src/libostree/ostree-repo-commit.c

```diff
@@ -2618,16 +2619,20 @@ write_directory_content_to_mtree_internal (OstreeRepo                  *self,
   else
     {
       guint64 file_obj_length;
-      const char *loose_checksum;
       g_autoptr(GInputStream) file_input = NULL;
       g_autoptr(GVariant) xattrs = NULL;
       g_autoptr(GInputStream) file_object_input = NULL;
       g_autofree guchar *child_file_csum = NULL;
       g_autofree char *tmp_checksum = NULL;
 
-      loose_checksum = devino_cache_lookup (self, modifier,
-                                            g_file_info_get_attribute_uint32 (child_info, "unix::device"),
-                                            g_file_info_get_attribute_uint64 (child_info, "unix::inode"));
+      /* only check the devino cache if the file info was not modified */
```

Comment:
What about xattrs?  For the rpm-ostree use case it'd make commit notably more expensive as suddenly we'd have to load and compare the xattrs.

If we're explicitly not comparing xattrs then let's at least add a comment.

Reply:
Ahh of course, we need to check xattrs too. Will add that.

Hmm, do you mean in the rpm-ostree local assembly path? Though we don't provide an xattr modifier when checking out files there, so we can probably do it in a smart way so we don't even bother comparing in such cases.

Reply:
Hmm...but actually in the layering case I think we *should* be setting the sepolicy to ensure we relabel any files created by scripts.  (In practice this probably doesn't matter much and really everything should mostly be `usr_t` but still).